File: test-pipeline.js
async function testPipeline() {
    console.log("🔍 TESTING LIQUID MILK BALLS INPUT PIPELINE");
    console.log("=" .repeat(80));
    
    const tests = [];
    
    // Test 1: Server is running
    console.log("\n1️⃣ Testing local server...");
    try {
        const response = await fetch('http://localhost:3000/healthz');
        if (response.ok) {
            console.log("  ✅ Server is responding");
            tests.push({name: "Server", status: "PASS"});
        } else {
            throw new Error(`Status: ${response.status}`);
        }
    } catch (error) {
        console.log("  ❌ Server not responding:", error.message);
        tests.push({name: "Server", status: "FAIL", error: error.message});
    }
    
    // Test 2: Publisher token generation
    console.log("\n2️⃣ Testing publisher token generation...");
    try {
        const response = await fetch('http://localhost:3000/api/publisher-token');
        const data = await response.json();
        console.log("  ✅ Publisher token generated");
        console.log("     Room:", data.room);
        console.log("     URL:", data.url);
        tests.push({name: "Publisher Token", status: "PASS"});
    } catch (error) {
        console.log("  ❌ Publisher token failed:", error.message);
        tests.push({name: "Publisher Token", status: "FAIL", error: error.message});
    }
    
    // Test 3: Viewer token generation
    console.log("\n3️⃣ Testing viewer token generation...");
    try {
        const response = await fetch('http://localhost:3000/api/viewer-token');
        const data = await response.json();
        console.log("  ✅ Viewer token generated");
        console.log("     Room:", data.room);
        console.log("     URL:", data.url);
        tests.push({name: "Viewer Token", status: "PASS"});
    } catch (error) {
        console.log("  ❌ Viewer token failed:", error.message);
        tests.push({name: "Viewer Token", status: "FAIL", error: error.message});
    }
    
    // Test 4: HTML files exist
    console.log("\n4️⃣ Testing HTML files...");
    const htmlFiles = ['publisher.html', 'td-input-viewer.html', 'return-viewer.html'];
    let htmlOk = true;
    for (const file of htmlFiles) {
        try {
            const response = await fetch(`http://localhost:3000/${file}`);
            if (response.ok) {
                console.log(`  ✅ ${file} exists`);
            } else {
                console.log(`  ❌ ${file} not found (${response.status})`);
                htmlOk = false;
            }
        } catch (error) {
            console.log(`  ❌ ${file} error:`, error.message);
            htmlOk = false;
        }
    }
    tests.push({name: "HTML Files", status: htmlOk ? "PASS" : "FAIL"});
    
    // Summary
    console.log("\n" + "=".repeat(80));
    console.log("📊 TEST SUMMARY");
    console.log("=".repeat(80));
    
    const passed = tests.filter(t => t.status === "PASS").length;
    const failed = tests.filter(t => t.status === "FAIL").length;
    
    tests.forEach(test => {
        const icon = test.status === "PASS" ? "✅" : "❌";
        console.log(`${icon} ${test.name}: ${test.status}`);
        if (test.error) {
            console.log(`   Error: ${test.error}`);
        }
    });
    
    console.log(`\n📈 Results: ${passed} passed, ${failed} failed`);
    
    if (failed === 0) {
        console.log("\n🎉 ALL TESTS PASSED! Pipeline is ready!");
        console.log("\n📋 NEXT STEPS:");
        console.log("   1. Open http://localhost:3000/publisher.html in Firefox");
        console.log("      - Allow camera access when prompted");
        console.log("      - Click 'Start Publishing'");
        console.log("   2. In TouchDesigner textport (Alt+T), run:");
        console.log("      execfile('C:/Users/krista-showputer/Desktop/liquid-milk-balls-web/auto_config_webclient.py')");
        console.log("   3. Look at the Web Client TOP - you should see video!");
    } else {
        console.log("\n⚠️ Some tests failed. Fix the issues above before proceeding.");
    }
}

testPipeline().catch(console.error);

File: test_pipeline.py
"""
Complete Pipeline Test Script
Tests all components of the video processing loop
"""
import requests
import time

def test_server():
    """Test if local server is running"""
    try:
        response = requests.get("http://localhost:3000/healthz", timeout=2)
        if response.status_code == 200:
            print("[OK] Server is running on port 3000")
            return True
        else:
            print(f"[FAIL] Server returned status {response.status_code}")
            return False
    except Exception as e:
        print(f"[FAIL] Server not responding: {e}")
        return False

def test_publisher_page():
    """Test if publisher page is accessible"""
    try:
        response = requests.get("http://localhost:3000/publisher.html", timeout=2)
        if response.status_code == 200:
            print("[OK] Publisher page is accessible")
            return True
        else:
            print(f"[FAIL] Publisher page returned {response.status_code}")
            return False
    except Exception as e:
        print(f"[FAIL] Publisher page error: {e}")
        return False

def test_viewer_page():
    """Test if return viewer page is accessible"""
    try:
        response = requests.get("http://localhost:3000/return-viewer.html", timeout=2)
        if response.status_code == 200:
            print("[OK] Return viewer page is accessible")
            return True
        else:
            print(f"[FAIL] Return viewer page returned {response.status_code}")
            return False
    except Exception as e:
        print(f"[FAIL] Return viewer page error: {e}")
        return False

def test_token_api():
    """Test if LiveKit token generation works"""
    try:
        response = requests.get("http://localhost:3000/api/processed-publisher-token", timeout=2)
        if response.status_code == 200:
            data = response.json()
            if 'token' in data and 'whipUrl' in data:
                print(f"[OK] Token API working")
                print(f"     WHIP URL: {data['whipUrl'][:80]}...")
                return True
            else:
                print(f"[FAIL] Token API missing fields: {data.keys()}")
                return False
        else:
            print(f"[FAIL] Token API returned {response.status_code}")
            return False
    except Exception as e:
        print(f"[FAIL] Token API error: {e}")
        return False

def main():
    print("="*60)
    print("PIPELINE HEALTH CHECK")
    print("="*60)
    
    results = {
        "Server": test_server(),
        "Publisher Page": test_publisher_page(),
        "Viewer Page": test_viewer_page(),
        "Token API": test_token_api()
    }
    
    print("\n" + "="*60)
    print("SUMMARY")
    print("="*60)
    
    for component, status in results.items():
        symbol = "[OK]" if status else "[FAIL]"
        print(f"{symbol} {component}")
    
    all_passed = all(results.values())
    
    print("\n" + "="*60)
    if all_passed:
        print("ALL SYSTEMS OPERATIONAL!")
        print("="*60)
        print("\nNext steps:")
        print("1. Open OBS")
        print("2. Enable WebSocket Server:")
        print("   - Tools > WebSocket Server Settings")
        print("   - [X] Enable WebSocket server")
        print("   - Leave password blank")
        print("   - Click OK")
        print("3. Click 'Start Streaming'")
        print("4. Open http://localhost:3000/publisher.html")
        print("5. Open http://localhost:3000/return-viewer.html")
    else:
        print("SOME SYSTEMS FAILED - Check errors above")
        print("="*60)

if __name__ == "__main__":
    main()
